Source/Core/VideoCommon/UberShaderPixel.cpp

```diff
@@ -150,15 +150,23 @@ ShaderCode GenPixelShader(APIType ApiType, const ShaderHostConfig& host_config,
   // Uniform index -> texture coordinates
   if (numTexgen > 0)
   {
+    for (u32 i = 0; i < numTexgen; i++)
+    {
+      out.Write("\tint2 fixpoint_uv{} = int2(", i);
+      out.Write("(tex{}.z == 0.0 ? tex{}.xy : tex{}.xy / tex{}.z)", i, i, i, i);
+      out.Write(" * " I_TEXDIMS "[{}].zw);\n", i);
+      // TODO: S24 overflows here?
```

Comment:
What's the story here?

Reply:
The same comment exists in specialized shaders:

https://github.com/dolphin-emu/dolphin/blob/4d37dad20d230b003b74e38ff4c5af68e6ed232e/Source/Core/VideoCommon/PixelShaderGen.cpp#L786-L792

Ubershaders already had (basically) the same code, just in a different place and without the comment.

The software renderer doesn't seem to worry about it at all:

https://github.com/dolphin-emu/dolphin/blob/4d37dad20d230b003b74e38ff4c5af68e6ed232e/Source/Core/VideoBackends/Software/Rasterizer.cpp#L228-L238

https://github.com/dolphin-emu/dolphin/blob/4d37dad20d230b003b74e38ff4c5af68e6ed232e/Source/Core/VideoBackends/Software/Rasterizer.cpp#L117-L123

It's something that might need further investigation, given the inconsistent results I got with my first attempt at hardware testing where I ended up overflowing really hard.  The [zww-water](https://fifo.ci/dff/zww-water/) test case is also incorrect in the software renderer so it might be wrong.  But those are something to come back to in the future.